### PR TITLE
OCPNODE-2553: add support for UserNamespaceLevel field

### PR DESCRIPTION
**Pull request by haircommander**
according to https://github.com/openshift/enhancements/blob/master/enhancements/kubelet/user-namespaces-support.md

### CODE REVIEW

**Review by ibihim — COMMENTED** [review]
Let me summarize and verify we are on the same page:
- We don't encode interdependence with other fields. We assume the person that defines the SCC doesn't set conflicting values.
- We don't set a default for `scc.UserNamespaceLevel`, therefore it joins the fields that aren't set by default like `hostNetwork` or more like `ReadOnlyRootFilesystem`, which doesn't enforce it if not set.

File: pkg/securitycontextconstraints/sccmatching/provider.go [hunk 1]
```diff
@@ -362,6 +366,15 @@ func (s *simpleProvider) ValidateContainerSecurityContext(pod *api.Pod, containe
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("hostIPC"), podSC.HostIPC(), "Host IPC is not allowed to be used"))
 	}
 
+	if s.scc.UserNamespaceLevel == securityv1.NamespaceLevelRequirePod {
```
[the remark this answers is not in the document]
Comment (haircommander) [reply pkg/securitycontextconstraints/sccmatching/provider.go:369]: I followed the other namespace fields here https://github.com/openshift/apiserver-library-go/pull/131/files#diff-f6878135c70949df461a2eb5c5c5e31a15bd494cf9bc0b544ad6a42af239b071R361-R367 so I'm not sure exactly but I followed convention

**Comment by haircommander** [comment]
> We don't encode interdependence with other fields. We assume the person that defines the SCC doesn't set conflicting values.

Correct, there's some validation the kubelet / API does, but we don't duplicate that here.

> We don't set a default for scc.UserNamespaceLevel, therefore it joins the fields that aren't set by default like hostNetwork or more like ReadOnlyRootFilesystem, which doesn't enforce it if not set.

yeah exactly

**Comment by deads2k** [comment]
> /lgtm

Please add yourself as an approver

/approve

**Pull request merged**